### PR TITLE
[WIP] feat!: updating theme provider and docs for personal theme – Kristóf’s branch

### PR DESCRIPTION
**Pull request by kripod-wise**
## Context

https://transferwise.atlassian.net/browse/DS-2677

As the part of our rebranding process, Neptune Web components’ colors have to be updated to reflect our new look and feel.

I’ve tried my best documenting my progress on a per-Figma-component basis here: https://transferwise.atlassian.net/wiki/spaces/DS/pages/2503181525/Web+redesign+progress

Additionally, I’ve proposed major improvements for the Neptune Web package ecosystem, detailed on a wiki page: https://transferwise.atlassian.net/wiki/spaces/WEB/pages/2493943945/Neptune+design+system+packages+Issues+Suggestions

## Changes

 

- Added personal theme styles for many components
- Extended `useTheme` hook to return whether the selected theme is modern or not
- Fixed minor issues that I came across while working on my primary task

## Considerations / Results

I’ve tried not to break existing markup and kept all the legacy styles intact.

## Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/actionButton/ActionButton.less [hunk 1]
```diff
@@ -27,4 +27,37 @@
       box-shadow: 0 0 0 4px var(--color-background-accent);
     }
   }
+
+  .np-theme-personal & {
+    color: var(--color-interactive-primary);
```
Comment (dozortsev) [inline packages/components/src/actionButton/ActionButton.less:32]: is it okey to use `-interactive-` token for text color? should we use only `-content-` for text?

File: packages/components/src/chips/Chip.less [hunk 2]
```diff
@@ -70,4 +70,42 @@
       color: white;
     }
   }
+
+  .np-theme-personal & {
+    font-weight: var(--font-weight-semi-bold);
```
Comment (dozortsev) [inline packages/components/src/chips/Chip.less:75]: typography styles for all components are done in another PR please don't fix typography as part of colors PR

File: packages/components/src/chips/Chip.less [hunk 3]
```diff
@@ -70,4 +70,42 @@
...
+    color: var(--color-interactive-primary);
+    padding: 0 var(--size-16);
```
Comment (dozortsev) [inline packages/components/src/chips/Chip.less:77]: padding/margin and other spacing styles needs to be done separately, as part of shape work

File: packages/components/src/common/bottomSheet/BottomSheet.less [hunk 4]
```diff
@@ -31,4 +31,8 @@
     // https://webkit.org/blog/7929/designing-websites-for-iphone-x/
     padding-bottom: env(safe-area-inset-bottom);
   }
+
+  .np-theme-personal & {
+    border-radius: var(--radius-large) var(--radius-large) 0 0;
```
Comment (dozortsev) [inline packages/components/src/common/bottomSheet/BottomSheet.less:36]: please leave shapes change for separate PR

File: packages/components/src/common/bottomSheet/BottomSheet.tsx [hunk 5]
```diff
@@ -166,8 +166,17 @@ const BottomSheet = (props: Props): ReactElement => {
       maxHeight: `calc(${screenHeight} - ${safeZoneHeight} - ${topbarHeight})`,
     };
   }
+
+  const theme = useTheme();
```
Comment (dozortsev) [inline packages/components/src/common/bottomSheet/BottomSheet.tsx:170]: BottomSheet started slide from right to left when opens up, instead of from bottom to top

not sure what causing this bug

https://user-images.githubusercontent.com/4015628/188632345-5704ab58-597e-422a-af19-846258e0e44f.mov

File: packages/components/src/accordion/AccordionItem/AccordionItem.js [hunk 6]
```diff
@@ -30,7 +30,7 @@ const AccordionItem = ({
         as="button"
         media={iconElement}
         title={title}
-        button={<Chevron orientation={open ? Position.TOP : Position.DOWN} />}
+        button={<Chevron orientation={open ? Position.TOP : Position.DOWN} size={Size.MEDIUM} />}
```
Comment (dozortsev) [inline packages/components/src/accordion/AccordionItem/AccordionItem.js:33]: in light theme size should be small, so I guess you need define size here based on theme (via `useTheme` hook)

File: packages/components/src/chips/Chip.tsx [hunk 7]
```diff
@@ -33,6 +34,10 @@ const Chip = ({
   const isActionable = onClick || onKeyPress;
   const defaultRole = isActionable ? 'button' : undefined;
   const tabIndex = isActionable ? 0 : -1;
+
+  const theme = useTheme();
+  const isThemeModern = useMemo(() => modernThemes.has(theme), [theme]);
```
Comment (dozortsev) [inline packages/components/src/chips/Chip.tsx:39]: I would instead make `useTheme()` more useful

```js
const { isPersonalThem, isLightTheme } = useTheme()
```

this way you will safe a tone of `if` and equality (`===`) statements inside Neptune web and for consumers of Neptune

Comment (dozortsev) [reply packages/components/src/chips/Chip.tsx:39]: we have similar thing do page direction data

https://github.com/transferwise/neptune-web/blob/main/packages/components/src/common/hooks/useDirection/useDirection.ts#L6-L10

File: stylelint.config.js [hunk 8]
```diff
@@ -24,7 +24,10 @@ module.exports = {
     'csstools/value-no-unknown-custom-properties': [
       true,
       {
-        importFrom: ['./node_modules/@transferwise/neptune-tokens/tokens.css'],
+        importFrom: [
+          './node_modules/@transferwise/neptune-tokens/tokens.css',
+          './node_modules/@transferwise/neptune-tokens/themes/personal/tokens.css',
```
Comment (dozortsev) [inline stylelint.config.js:29]: you should not need define `personal/tokens.css` here as tokens names are the same (only values are different) that is what it matters for this Stylelint plugin

File: packages/components/src/actionButton/ActionButton.less (hunk 1 above)
Comment (kripod-wise) [reply packages/components/src/actionButton/ActionButton.less:32]: I’ve used [the component remapping specs](https://www.figma.com/file/GO6vjcaEIukCbRdZ2yZ1Ms/VE---Components-remapping-specs?node-id=65%3A21950) as a reference, as instructed within the [Jira ticket](https://transferwise.atlassian.net/browse/DS-2677) my work is based upon.

File: packages/components/src/common/bottomSheet/BottomSheet.tsx (hunk 5 above)
Comment (kripod-wise) [reply packages/components/src/common/bottomSheet/BottomSheet.tsx:170]: This change shouldn’t affect that. If I remember correctly, the bottom sheet came in from the left even before this change, but I may be mistaken. and we should see whether this behavior is the same on the original `DS-2378-personal-theme` branch.

File: packages/components/src/chips/Chip.tsx (hunk 7 above)
Comment (kripod-wise) [reply packages/components/src/chips/Chip.tsx:39]: I’ve tried not to introduce any breaking changes within this PR, so that app owners who rely upon this package could upgrade to the latest version of Neptune Web without hassle, introducing no breaking changes throughout the flow.

However, if we really want return an object from `useTheme()`, I could refactor the code to do so.

File: stylelint.config.js (hunk 8 above)
Comment (kripod-wise) [reply stylelint.config.js:29]: We did introduce some new colors, as my IDE’s complaints led me to make this change. E.g. [`--color-interactive-accent` has been replaced by `--color-interactive-primary`](https://transferwise.atlassian.net/wiki/spaces/VR/pages/2437844264/Colour+tokens+old+vs+new#Interactive-colours).

File: packages/components/src/common/bottomSheet/BottomSheet.tsx (hunk 5 above)
Comment (dozortsev) [reply packages/components/src/common/bottomSheet/BottomSheet.tsx:170]: you can always double check on our prod 🙂

https://transferwise.github.io/neptune-web/components/overlays/BottomSheet

there it opens from bottom to top

File: packages/components/src/chips/Chip.tsx (hunk 7 above)
Comment (dozortsev) [reply packages/components/src/chips/Chip.tsx:39]: what breaking change will be introduced the thing that I suggested https://github.com/transferwise/neptune-web/pull/1585#discussion_r963647766?

I don't see any 🤔

File: stylelint.config.js (hunk 8 above)
Comment (dozortsev) [reply stylelint.config.js:29]: do you mean that `--color-interactive-primary` isn't defined in `neptune-tokens/tokens.css`? if so than this is a bug in neptune-tokens package

File: packages/components/src/common/bottomSheet/BottomSheet.tsx (hunk 5 above)
Comment (kripod-wise) [reply packages/components/src/common/bottomSheet/BottomSheet.tsx:170]: Yep, but I’d assume the production branch isn’t based upon `DS-2378-personal-theme`, on which my branch is based.

Comment (kripod-wise) [reply packages/components/src/common/bottomSheet/BottomSheet.tsx]: It turns out that the base `DS-2378-personal-theme` branch has the same issue. Therefore, further investigation should be done over there, as this isn’t the result of this PR.

**Review by richardmccartney — APPROVED** [review]
